code/asteroid/asteroid.cpp

```diff
@@ -732,25 +746,21 @@ void asteroid_create_asteroid_field(int num_asteroids, int field_type, int aster
 	Asteroid_field.speed = (float)asteroid_speed;
 	Asteroid_field.debris_genre = DG_ASTEROID;
 
-	for (int j = 0; j < MAX_ACTIVE_DEBRIS_TYPES; j++) {
-			Asteroid_field.field_debris_type[j] = -1;
-	}
+	Asteroid_field.field_debris_type.clear();
 
-	for (int j = 0; j < NUM_ASTEROID_SIZES; j++) {
-			Asteroid_field.field_asteroid_type[j] = false;
-	}
+	Asteroid_field.field_asteroid_type.clear();
 
 	int count = 0;
 	if (brown) {
-		Asteroid_field.field_asteroid_type[0] = true;
+		Asteroid_field.field_asteroid_type.push_back("Brown");
 		count++;
 	}
 	if (blue) {
-		Asteroid_field.field_asteroid_type[1] = true;
+		Asteroid_field.field_asteroid_type.push_back("Blue");
 		count++;
 	}
 	if (orange) {
-		Asteroid_field.field_asteroid_type[2] = true;
+		Asteroid_field.field_asteroid_type.push_back("Orange");
```

Comment:
You refer to these retail names a few times, here, in missionparse, etc.
At that point, these should probably become defines or something.

Reply:
That's a good idea. The sexps are going to need quite a bit of rewrite to handle this new code, so I'll punt on this for when I redo those in a follow-up.

Reply:
I see, fair enough.